File: Stock Data Clean.R
# Get the real map of China
library(jsonlite)
library(rjson)
library(tidyverse)
library(lubridate)
load("~/Desktop/PhDwork/Py-Github/Piao/IBM.Rdata")
# read.json
df=read_json("300435.txt")
x=df$data
y=x$klines
name=x$name
# get data
dfprice=c()
for (i in 1:length(y)) {
  id=i
  value=y[[i]]
  yy=tibble(id=id,value=value)
  dfprice=rbind(dfprice,yy)
}
# rename data
df=dfprice %>% select(-id) %>% 
  separate(value,c("time","kaipan","shoupan","high","low","tradenumber","trademoney","zhenfu","todayindex","todaydiff","changerate")
           , sep = "([,])")
# remove
rm(i,yy,x,y,dfprice)

write.csv(df,"df.csv")
df=read.csv("df.csv",header = T) %>% as_tibble() %>% 
  mutate(time=as.Date(time)) %>% 
  arrange(desc(time))

# remove file
file.remove("df.csv")


## plot
library(sysfonts)
library(showtextdb)
library(showtext)

showtext_auto()
ggplot(df)+
  geom_point(aes(time,low))+
  geom_line(aes(time,low))+
  labs(title = name)


df1=df %>% filter(time>as.Date("2021-03-18"))
df2=df %>% filter(time>as.Date("2021-03-18"))
ggplot(df1)+
  geom_point(aes(time,low))+
  geom_line(aes(time,low))+
  geom_line(data=df2,aes(time,high,color="red"))+
  labs(title = name)+ 
  geom_hline(yintercept=0.5*max(df$kaipan), linetype="dashed", color = "red", size=1)+
  geom_hline(yintercept=0.5*max(df$shoupan), linetype="dashed", color = "blue", size=1)

File: Stock Data clean.py
import pandas as pd
import numpy as np
import json  

# load data
f =open('/Users/Anderson/Desktop/PhDwork/Github-New/Stock_github/603260.txt',encoding='utf-8') #打开‘product.json’的json文件
res=f.read()  #读文件
print(json.loads(res))#把json串变成python的数据类型：字典   


a=json.loads(res)
type(a)# a里面包含字典信息；字典信息里面的data存储了股票每天的股价信息

# extract data
y=a['data'] 
xx=y['klines']# 股票每天的股价信息
type(xx)


#现在将list信息转成 Pd dataframe
df=pd.DataFrame(y)
df

#现在将 pd dataframe 的字符串分列
df=pd.DataFrame(y)
df[['time','kaipan','shoupan','max','min','percentage','percentage_value','volume','money','amplitude','turnover']]=df['klines'].str.split(',', expand=True)
print(df.tail())

#save data

stockname=df['code']
x= stockname.drop_duplicates()
print(x)
#xx.to_csv()

df.to_csv(x[0]+'.csv')






